docs/tabpy-tools.md

```diff
@@ -136,6 +137,95 @@ client.remove('WillItDefault')
+
+```
+"Successfully deployed PCA"
+```
+
+If you would like to deploy additional models using the deploy script, you can
+copy any python file to the ./models/scripts directory and modify setup.py to
+include all necessary packages when installing dependencies or alternatively install
+all the required dependencies manually.
+
+#### Principal Component Analysis (PCA)
+
+[Principal component analysis](https://en.wikipedia.org/wiki/Principal_component_analysis)
+is a statistical technique which extracts new, linearly uncorrelated, variables out of a
+dataset which capture the maximum variance in the data. In this way, PCA can be used to
+reduce the number of variables in a high dimensional dataset, a process that is called
+dimensionality reduction. The first principal component captures the largest amount of
+variance, while the second captures the largest portion of the remaining variance while
+remaining orthogonal to the first and so on. This allows the reduction of the number of
+dimensions while maintaining as much of the information from the original data as possible.
+PCA is useful in exploratory data analysis because complex relationships can be visualized
+in a 2D scatter plot of the first few principal components.
+
+TabPy’s implementation of PCA uses the scikit-learn [decomposition.PCA](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html)
+algorithm, which is further documented [here](https://scikit-learn.org/stable/modules/decomposition.html#pca). 
+It requires the selected component to be > 0 and <= number of variables you pass in to the function. 
+When passing categorical variables we perform the scikit-learn [One Hot Encoding](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html)
+to transform your non-numeric variables into a table of 0s and 1s. In order for One Hot
+Encoding to be performant we have limited the number of unique values your categorical
+column may contain to 25 and do not permit any nulls or empty strings in the column. Before
+PCA is performed, all variables are normalized to have a mean of 0 and unit variance using 
+the scikit-learn [StandardScaler](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.StandardScaler.html).
+
+A Tableau calculated field to perform PCA will look like
+
+```python
+
+tabpy.query(‘PCA’, 1, _arg1, _arg2, _arg3)[‘response’]
+
+```
+
+#### Sentiment Analysis
+
+[Sentiment analysis](https://en.wikipedia.org/wiki/Sentiment_analysis) is a technique
+which uses natural language processing to extract the emotional positivity or negativity
+– the sentiment – behind a piece of text and converts that into a numeric value. Our
+implementation of sentiment analysis returns a polarity score between -1 and 1 which rates
+the positivity of the string with 1 being very positive and -1 being very negative. Calling
+the Sentiment Analysis function from TabPy in Tableau will look like the following, where _arg1
```

Comment:
FYI all underscore characters to not make test italic need to be prefixed with \